### PR TITLE
Unflake the`dropped_client_doesnt_create_duplicate_carbons` test

### PR DESCRIPTION
**Pull request by gustawlippa**
The description of https://github.com/esl/MongooseIM/pull/4423 was not entirely correct, and this PR should unlflake the tests for good. The reason why the tests are changed, and not the implementation is because the underlying race condition is a bit tricky to fix elegantly.

`mod_carboncopy:remove_connection` is a handler for the `unset_presence` hook, and it removes CC state from the `ejabberd_sm` `info` field in the session record. This is however done in an async manner by the C2S process, and when the user terminates, the request is never handled. This doesn't matter much, as the user process exits shortly, and is removed from `ejabberd_sm` altogether. However, if they receives a message in this short time, CC might run, and see the process still with the CC state in the `ejabberd_sm` `info` field. This results in a CC copy sent on behalf of this exiting process. Since a fix would be overly complicated, and to proceed with unflaking the tests, the test will now wait for C2S process exit.

GH Actions run manually: https://github.com/esl/MongooseIM/actions/runs/12318614146/job/34383742037

### CODE REVIEW

**Comment by mongoose-im** [comment]
[elasticsearch_and_cassandra_27](https://circleci.com/gh/esl/MongooseIM/250941) / elasticsearch_and_cassandra_mnesia / 4d7b462dec6a45494910c95286aaf63925578f7e
Reports [root](http://esl.github.io/circleci-mim-results/s3_reports.html?prefix=PR/4437/250941/elasticsearch_and_cassandra_mnesia.27.1.2-amd64)/ [big](https://esl.github.io/html-zip-reader/PR/4437/250941/elasticsearch_and_cassandra_mnesia.27.1.2-amd64/big.tar.gz//ct_run.test%401c1dbfa2a3f7.2024-12-13_15.41.48/index.html)
**OK: 472** / Failed: 0 / **User-skipped: 49** / Auto-skipped: 0

**Comment by gustawlippa** [comment]
The `setup` CI job turned red because I accidentally rerun the CircleCI job after a long time I believe - there is no issue with CircleCI, and the tests have passed in reality.

File: big_tests/tests/carboncopy_SUITE.erl [hunk 1]
```diff
@@ -180,13 +180,20 @@ dropped_client_doesnt_create_duplicate_carbons(Config) ->
...
+              C2SPid = mongoose_helper:get_session_pid(Alice2),
               escalus_client:stop(Config, Alice2),
               escalus:assert(is_presence_with_type, [<<"unavailable">>],
                              escalus_client:wait_for_stanza(Alice1)),
+              %% Ensure there is no session so that the test doesn't flake and fail by chance.
+              %% The issue comes from the fact that mod_presence sends the "unavailable" stanza
+              %% just before removing the is removed from ejabberd_sm. If the msg is received by
+              %% mod_carboncopy in this short window, it still considers the resource enabled.
+              %% It's a race condition, but from the user's perspective it shouldn't be a big issue.
+              mongoose_helper:wait_for_pid_to_die(C2SPid),
```
Comment (NelsonVides) [inline big_tests/tests/carboncopy_SUITE.erl:192]: That's a fine one, we've used this trick to ensure we wait for the pid to die many other times 👌🏽

**Review by NelsonVides — APPROVED** [review]
LGTM 💪🏽

**Pull request merged**